src/roman.rs

```diff
@@ -91,25 +91,27 @@ impl Roman {
     /// cannot be appropriately formatted using the seven standard numerals.
     pub fn new(n: u16) -> Option<Roman> {
         match n {
-            n @ 1..=4999 => Some(Roman(n)),
+            n if n <= 4999 => NonZeroU16::new(n).map(Roman),
             _ => None,
         }
     }
 
-    /// Creates a `Roman` value based on a `u16`.
+    /// Creates a `Roman` value based on a [`NonZeroU16`].
     ///
-    /// This function will return any `u16` wrapped in a `Roman` newtype
+    /// This function will return any [`NonZeroU16`] wrapped in a `Roman` newtype
```

Comment:
`NonZeroU16`  is a weird type that I doubt most people ever use. As a result, I don't believe it should really be a part of the public interface. Not sure where to say that, but this seems as good a place as any: I think that neither the constructors nor the accessor ("to_inner" or whatever) should present a `NonZeroU16` instead of just a straight up `u16`.

What do you think? Is there some advantage to exposing the underlying funny type?

Reply:
The advantage is that when you end up using `NonZero*`, you won't be forced to use weird conversions like `NonZeroU16::new(...).expect("It can never be 0")`.

`NonZero*` types are not used frequently, but sometimes they are used anyway.

Reply:
(And using `u16` in `new_unchecked` would turn it into an `unsafe` function)

Reply:
The issue is that a library like this is created either to be silly (my reason) or for convenience (...presumably the reason someone would use it instead of just reimplementing it), so including an inconvenient type is kind of defeating the purpose, right? The latest version of the code deletes that constructor entirely for the reasons given above: a convenience method isn't useful if it isn't convenient.

It might make sense to offer access to the underlying non-zero value as well as a normal version of it. `get` for one and `into_inner` for the other?

Reply:
Hm, I didn't notice the latest commit.

I wouldn't say that `new_unchecked` is useless here - it is convenient when you sure about the value you pass in. This may remove some `unwrap`s.

> It might make sense to offer access to the underlying non-zero value as well as a normal version of it. get for one and into_inner for the other?

Yep, that's what I've said in the [next thread](https://github.com/archer884/xvii/pull/10#discussion_r502907641). For me "value" makes more sense, than "get", but it's just naming.

Reply:
I'm good with `value()`. I used `get()` because that was the name of the method already on `NonZeroU16`, but I kinda like `value()` better.